### PR TITLE
gh-1743 Sending simple eth transaction

### PR DESCRIPTION
**Pull request by DmitryBespalov**
Handles #1743 

Pair programmed with @biafra23 

Changes proposed in this pull request:
- RLP encoding / decoding
- integration test sending native coin on Ethereum Rinkeby
  - Ethereum JSON RPC types (Quantity, Data) using `EthRpc1.Quantity` and `EthRpc1.Data`
  - Using the types in the EthRpc1 API methods
  - Ether Units and conversion (Wei, Gwei, ..., Ether) using `Eth.Amount`
  - Estimating transaction, getting nonce, and gas price in one call (batch request)
  - Using test infura key in the tests

### CODE REVIEW

**Review by sche — APPROVED** [review]

File: Packages/Ethereum/Tests/JsonRpc2Tests/AsyncClientTests.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ import TestHelpers
 class AsyncClientTests: XCTestCase {
 
     let client = JsonRpc2.AsyncClient(
-        transport: JsonRpc2.AsyncHTTPTransport(url: "https://mainnet.infura.io/v3/PROJECTID"),
+        transport: JsonRpc2.AsyncHTTPTransport(url: "https://mainnet.infura.io/v3/fda31d5c85564ae09c97b1b970e7eb33"),
```
Comment (MouazAlzahabi) [inline Packages/Ethereum/Tests/JsonRpc2Tests/AsyncClientTests.swift:17]: I think we should not put the project id hardcoded here as it's open source code

File: Packages/Ethereum/Sources/Ethereum/Rlp.swift [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+        let items = try zip(decodedArray, prototypeArray).map { decodedValue, itemPrototype in
+            try itemPrototype.decode(value: decodedValue, coder: coder)
+        }
+
+        return items
+    }
+}
+
+//
+//public func == (lhs: RlpCodable, rhs: RlpCodable) -> Bool {
```
Comment (MouazAlzahabi) [inline Packages/Ethereum/Sources/Ethereum/Rlp.swift]: Why we have this commented code?

Comment (DmitryBespalov) [reply Packages/Ethereum/Sources/Ethereum/Rlp.swift]: forgotten code

File: Packages/Ethereum/Tests/JsonRpc2Tests/AsyncClientTests.swift (hunk 1 above)
Comment (DmitryBespalov) [reply Packages/Ethereum/Tests/JsonRpc2Tests/AsyncClientTests.swift:17]: this is a special testing infura id that we intended to keep public in the test code. It's a throwaway id.

**Review by MouazAlzahabi — APPROVED** [review]

**Pull request merged**